include/zephyr/net/wifi_mgmt.h

```diff
@@ -119,6 +129,20 @@ struct wifi_status {
 	int status;
 };
 
+struct wifi_iface_status {
+	int state; /* enum wifi_iface_state */
```

Comment:
Why don't we use `enum wifi_iface_state` instead of `int`?

Reply:
Typically, `int` is used in definition for enum's to avoid ambiguity as some compilers/systems use `char/int` for enums. This is mainly for the interface boundary files (host to chip) as they use different compilers/systems.

In this case, I can use `enum`.

Reply:
So, two options: 

1. use `int` and return `-1` for unknown
2. use `enum` and return `*_UNKNOWN`

I prefer #1 but okay to switch to #2 . Any hard preferences?

Reply:
Usage would be `enum`, but storage would be `int` to avoid incompatibility, but I can change to use `enum` for storage as well, it should be okay for now.